### PR TITLE
Search architecture docs

### PR DESCRIPTION
**Pull request by abhinavdangeti**
First stab at ascii docs.

### CODE REVIEW

File: modules/learn/pages/services-and-indexes/services/search-service.adoc [hunk 1]
```diff
@@ -16,6 +16,28 @@ These include:
 This keeps result-sets manageably small, even when the total number of documents returned is extremely large.
 * _Fast indexes_, which support a wide range of possible text-searches.
 
+== Search Service Architecture
+
 The Search Service requires the Data Service, but does not require, and has no relation to the Index Service or Query Service.
```
Comment (amarantha-k) [inline modules/learn/pages/services-and-indexes/services/search-service.adoc]: With N1QL FTS integration, does the statement "has no relation to the Index service or Query Service" need rewording?

File: modules/learn/pages/services-and-indexes/services/search-service.adoc [hunk 2]
```diff
@@ -16,6 +16,28 @@ These include:
...
 
+[#search_service_architecture]
+image::services-and-indexes/services/searchServiceArchitecture.png[,700,align=left]
+
+The illustration depicts the following:
+
+* *Data Service*: Uses the DCP protocol to stream data-mutations as batches from the producer to multiple consumers instantiated across the FTS nodes.
+* *FTS Service*: When a FTS index is created, the number of partitions chosen for the index will directly represent the number of DCP consumers created across all FTS nodes. Each index partition will cover a certain number of the bucket partitions. For example, if the number of bucket partitions were 120 (vbuckets), and the number of index partitions chosen were 6, each index partition will hold data from 20 vbuckets.
```
Comment (amarantha-k) [inline modules/learn/pages/services-and-indexes/services/search-service.adoc]: rename all "FTS" to "Search" or "full-text search"

Comment (amarantha-k) [inline modules/learn/pages/services-and-indexes/services/search-service.adoc]: We recommend using active voice and present tense as much as possible.

Comment (abhinavdangeti) [reply modules/learn/pages/services-and-indexes/services/search-service.adoc]: Got it, let me re-word this bit then.

Comment (abhinavdangeti) [reply modules/learn/pages/services-and-indexes/services/search-service.adoc]: Will do.

File: modules/learn/pages/services-and-indexes/services/search-service.adoc (hunk 1 above)
Comment (abhinavdangeti) [reply modules/learn/pages/services-and-indexes/services/search-service.adoc]: Ah good point, let me re-think it a bit.

**Comment by amarantha-k** [comment]
@abhinavdangeti Looks good to me. Do you have any pending changes for this PR? If not, I think we should merge the changes to fts-updates-1.

**Comment by abhinavdangeti** [comment]
Hey @amarantha-k , I'll push up the next set of changes as a separate PR. And yes - I've made the changes on top of the fts-updates-1 branch.

**Comment by amarantha-k** [comment]
Great! Thanks for confirming. I'll merge this PR right away.

**Pull request merged**